src/main/java/org/example/httpCrawler/WebCrawler.java

```diff
@@ -0,0 +1,95 @@
+package org.example.httpCrawler;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.io.File;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.net.http.HttpHeaders;
+import java.util.Scanner;
+import java.util.Map;
+import java.util.HashMap;
+
+@Slf4j
+public class WebCrawler {
+    private static final HttpClient client = HttpClient.newHttpClient();
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    public static void main(String[] args) throws IOException {
+        Scanner scanner = new Scanner(System.in);
+        System.out.println("Input URL: ");
+        String inputUrl = scanner.nextLine();
+        scanner.close();
+
+        if (!isValidURL(inputUrl)) {
+            log.warn("Invalid URL: {}", inputUrl);
+            return;
+        }
+
+        try {
+            crawl(inputUrl);
+        } catch (Exception e) {
+            log.error("Error when handling URL: {}", e.getMessage());
+        }
+    }
+
+    private static boolean isValidURL(String url) {
+        try {
+            new URI(url).toURL();
```

Comment:
Validation can be echanced :        String scheme = uri.getScheme();
        return scheme != null && (scheme.equals("http") || scheme.equals("https"));